app/mailers/admin_mailer.rb

```diff
@@ -5,4 +5,12 @@ def threshold_email_reminder(admin_users, petitions)
     @petitions = petitions
     mail(subject: "Petitions alert", to: admin_users.map(&:email))
   end
+
+  def petitions_report
+    @petitions = Petition.recently_opened_or_signed
+
+    subject = "E-petitions report #{(Date.current - 1.week).strftime("%d %b %Y")} - #{Date.current.strftime("%d %b %Y")}"
```

Comment:
I would probably inject these values as default args, e.g.

``` ruby
def petitions_report(now = Time.current.beginning_of_day)
  since = 1.week.since(now)
  subject = "Petitions report #{since.strftime("%d %b %Y")} - #{now.strftime("%d %b %Y")}"

  @petitions = Petition.recently_opened_or_signed

  mail(subject: subject, to: Site.report_email)
end
```

I'd probably also change the `recently_opened_or_signed` to `opened_or_signed_within` and pass those values into that method.

Reply:
Thanks. Updated [class method](https://github.com/StatesOfJersey/e-petitions/pull/38/commits/1eb38a4384e8d48d3c4477d1fbdd1c35d8a26cb2#diff-c7ed985e8cec2507095976e6089841dcR338) and [mailer](https://github.com/StatesOfJersey/e-petitions/pull/38/commits/1eb38a4384e8d48d3c4477d1fbdd1c35d8a26cb2#diff-b4ee7a88831b433a12d19b3586fb31d7)